frontend/src/components/editor/chrome/wrapper/pending-ai-cells.tsx

```diff
@@ -0,0 +1,125 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+
+import { useAtomValue, useStore } from "jotai";
+import { ChevronDown, ChevronUp, SparklesIcon } from "lucide-react";
+import { useState } from "react";
+import { Button } from "@/components/ui/button";
+import { stagedAICellsAtom, useStagedCells } from "@/core/ai/staged-cells";
+import type { CellId } from "@/core/cells/ids";
+import { cn } from "@/utils/cn";
+import {
+  AcceptCompletionButton,
+  RejectCompletionButton,
+} from "../../ai/completion-handlers";
+import { acceptStagedCell, rejectStagedCell } from "../../cell/StagedAICell";
+import { scrollAndHighlightCell } from "../../links/cell-link";
+
+export const PendingAICells: React.FC = () => {
+  const [currentIndex, setCurrentIndex] = useState<number | null>(null);
+
+  const stagedAiCells = useAtomValue(stagedAICellsAtom);
+  const listStagedCells = [...stagedAiCells.keys()];
+  const store = useStore();
+  const { deleteStagedCell, removeStagedCell } = useStagedCells(store);
+
+  if (stagedAiCells.size === 0) {
+    return null;
+  }
+
+  const scrollToCell = (cellId: CellId) => {
+    scrollAndHighlightCell(cellId, "focus");
+  };
+
+  const clickUp = () => {
+    if (currentIndex === null) {
+      setCurrentIndex(0);
+      scrollToCell(listStagedCells[0]);
+      return;
+    }
+
+    if (currentIndex > 0) {
+      setCurrentIndex(currentIndex - 1);
+      scrollToCell(listStagedCells[currentIndex - 1]);
+      return;
+    }
+
+    // Wrap around to last cell
+    setCurrentIndex(listStagedCells.length - 1);
+    scrollToCell(listStagedCells[listStagedCells.length - 1]);
+  };
```

Comment:
this can be simplified a bit:

```ts
const clickUp = () => {
  const newIndex = ((currentIndex ?? 0) - 1 + listStagedCells.length) % listStagedCells.length;
  setCurrentIndex(newIndex);
  scrollToCell(listStagedCells[newIndex]);
};

const clickDown = () => {
  const newIndex = ((currentIndex ?? 0) + 1) % listStagedCells.length;
  setCurrentIndex(newIndex);
  scrollToCell(listStagedCells[newIndex]);
};
```

maybe you'd want to put the complex line in a util as well